buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -59,6 +59,12 @@ protected CharBuffer initialValue() throws Exception {
             return CharBuffer.allocate(1024);
         }
     };
+    private static final FastThreadLocal<byte[]> BYTE_ARRAYS = new FastThreadLocal<byte[]>() {
+        @Override
+        protected byte[] initialValue() throws Exception {
+            return PlatformDependent.allocateUninitializedArray(1024);
```

Comment:
fwiw gRPC does something similar, but we wrap the buffer in a weak ref to avoid extra memory.  It prolly doesn't matter for 1k chunks, but you might consider it.

Reply:
@carl-mastrangelo yeah I think it not matters here... anyway do you have a link handy ?

Reply:
https://github.com/grpc/grpc-java/blob/v1.15.1/protobuf-lite/src/main/java/io/grpc/protobuf/lite/ProtoLiteUtils.java#L180


In our case we can go up to about 4MB, but I guess that doesn't matter much.

Reply:
Thanks... but yeah I think we should be good here without the need of doing this. If we ever want to use bigger `byte[]` we can re-investigate.